torchrl/collectors/collectors.py

```diff
@@ -315,6 +317,43 @@ def __len__(self) -> int:
             return -(self.total_frames // -self.requested_frames_per_batch)
         raise RuntimeError("Non-terminating collectors do not have a length")
 
+    @classmethod
+    def from_policy_factory(
```

Comment:
@mikaylagawarecki 

This is a bit convoluted.

We could also have one more kwarg in the constructors, but since there are many kwargs there already, and many collector subclasses, IDK if that's the most economical solution.

Happy to make this a kwargs that is exclusive with the `policy` arg if you think it's more suited

Reply:
I think this is ok

Just a question -- do you think that giving the CustomCollectorCls a  __name__ that depended on its cls would be helpful for debugging, or would that make it harder to find where the wrapper class was defined 😅